### PR TITLE
Made some fixes to Listener

### PR DESCRIPTION
**Pull request by TAImatem**

### CODE REVIEW

File: OWML.Launcher/SocketListener.cs [hunk 1]
```diff
@@ -11,7 +11,8 @@ namespace OWML.Launcher
 {
     public class SocketListener
     {
-        private const int BufferSize = 1024;
+        private const string Separator = "\n--------------------------------";
+        private const int BufferSize = 1<<18;
```
Comment (misternebula) [inline OWML.Launcher/SocketListener.cs:15]: Either add a code comment here or write this out plainly - I can see someone (or us) having to look up what this is equal to

File: OWML.Launcher/SocketListener.cs [hunk 2]
```diff
@@ -73,23 +78,40 @@ private void ListenToSocket()
...
 
         private void ProcessMessage(byte[] bytes, int count)
         {
-            var json = Encoding.UTF8.GetString(bytes, 0, count);
-
-            var data = JsonConvert.DeserializeObject<SocketMessage>(json);
-
-            if (data.Type == MessageType.Quit)
+            var message = Encoding.UTF8.GetString(bytes, 0, count);
+            var jsons = message.Split('\n');
```
Comment (misternebula) [inline OWML.Launcher/SocketListener.cs:89]: What if a message inside a json contains a \n

Comment (TAImatem) [reply OWML.Launcher/SocketListener.cs:89]: it shouldn't

Comment (TAImatem) [reply OWML.Launcher/SocketListener.cs:89]: inside messages themselves they are written as "\\n" which != '\n', and jsons are one-liners

Comment (TAImatem) [reply OWML.Launcher/SocketListener.cs:89]: so, unless you add newlines into json yourself it will work fine 😝

Comment (TAImatem) [reply OWML.Launcher/SocketListener.cs:89]: and I considered secret combination to be a bit iffy

Comment (amazingalek) [reply OWML.Launcher/SocketListener.cs:89]: > inside messages themselves they are written as "\\n"

Always? Whether log came from OWML or Unity?

File: OWML.Launcher/SocketListener.cs [hunk 3]
```diff
@@ -73,23 +78,40 @@ private void ListenToSocket()
...
+            foreach (var json in jsons)
             {
-                Environment.Exit(0);
-            }
+                if (json == "")
```
Comment (amazingalek) [inline OWML.Launcher/SocketListener.cs:92]: `string.IsNullOrEmpty(json)`, maybe?

File: OWML.Launcher/SocketListener.cs (hunk 2 above)
Comment (TAImatem) [reply OWML.Launcher/SocketListener.cs:89]: yes, it does that when converting to json string (and yes I tested it)

File: OWML.Launcher/SocketListener.cs [hunk 4]
```diff
@@ -11,7 +11,8 @@ namespace OWML.Launcher
 {
     public class SocketListener
     {
-        private const int BufferSize = 1024;
+        private const string Separator = "\n--------------------------------";
+        private const int BufferSize = 262144;
```
Comment (amazingalek) [inline OWML.Launcher/SocketListener.cs:15]: Sure this won't clog up *something*?

File: OWML.Launcher/SocketListener.cs (hunk 2 above)
Comment (amazingalek) [reply OWML.Launcher/SocketListener.cs:89]: If you say so ;)

Comment (TAImatem) [reply OWML.Launcher/SocketListener.cs:89]: as it works with both OWML message (one from Owo about changed settings) and Unity message (when testing with stack overflow trace)

File: OWML.Launcher/SocketListener.cs (hunk 4 above)
Comment (TAImatem) [reply OWML.Launcher/SocketListener.cs:15]: worked fine with stack overflow trace (~90-100k symbols)

File: OWML.Launcher/SocketListener.cs (hunk 3 above)
Comment (TAImatem) [reply OWML.Launcher/SocketListener.cs:92]: OK

Comment (TAImatem) [reply OWML.Launcher/SocketListener.cs:92]: though I'm gonna use IsNullOrWhiteSpace instead 😉

**Comment by amazingalek** [comment]
Looks good but I'll let @misternebula have the final word ;)

**Pull request merged**